spec/system/support_interface/reasons_for_rejection_dashboard_spec.rb

```diff
@@ -0,0 +1,192 @@
+    application_form = create(:application_form)
+    create(:application_choice, application_form: application_form)
+    references = create_list(:reference, 2, application_form: application_form)
+    references.each { |reference| CandidateInterface::RequestReference.new.call(reference) }
+    [application_form, references]
+  end
+
+  def provide_references(references)
+    references.each { |reference| SubmitReference.new(reference: reference, send_emails: false).save! }
+  end
+
+  def reject_application_with_structured_reasons(application_choice)
+    application_choice.update!(
+      status: :rejected,
+      structured_rejection_reasons: {
+        course_full_y_n: 'Yes',
+        candidate_behaviour_y_n: 'Yes',
+        honesty_and_professionalism_y_n: 'Yes',
+        performance_at_interview_y_n: 'Yes',
+        qualifications_y_n: 'Yes',
+        quality_of_application_y_n: 'Yes',
+        safeguarding_y_n: 'Yes',
+        offered_on_another_course_y_n: 'Yes',
+        interested_in_future_applications_y_n: 'Yes',
+        other_advice_or_feedback_y_n: 'Yes',
+      },
+      rejected_at: Time.zone.now,
+    )
+  end
+
+  def reject_application_without_structured_reasons(application_choice)
+    application_choice.update!(
+      status: :rejected,
+      rejected_at: Time.zone.now,
+    )
+  end
+
+  def and_there_are_candidates_and_application_forms_in_the_system
+    allow(EndOfCycleTimetable).to receive(:apply_reopens).and_return(60.days.ago)
+    Timecop.freeze(@today - 50.days) do
+      @application_form1, @references1 = create_application_form_with_references
+      create(:authentication_token, user: @application_form1.candidate, hashed_token: '0987654321')
+    end
+
+    Timecop.freeze(@today - 40.days) do
+      provide_references(@references1)
+      reject_application_with_structured_reasons(@application_form1.application_choices.first)
+    end
+
+    Timecop.freeze(@today - 2.days) do
+      @application_form2, @references2 = create_application_form_with_references
+      @application_form3, @references3 = create_application_form_with_references
+      create(:authentication_token, user: @application_form2.candidate, hashed_token: '9876543210')
+      create(:authentication_token, user: @application_form3.candidate, hashed_token: '8765432109')
+    end
+
+    Timecop.freeze(@today) do
+      provide_references(@references2)
+      provide_references(@references3)
+      reject_application_with_structured_reasons(@application_form2.application_choices.first)
+      reject_application_without_structured_reasons(@application_form3.application_choices.first)
+    end
+  end
+
+  def when_i_visit_the_performance_page_in_support
+    visit support_interface_performance_path
+  end
+
+  def and_i_click_on_the_reasons_for_rejection_metrics_link
+    click_on 'Reasons for rejection'
+  end
+
+  def then_i_should_see_reasons_for_rejection_course_full
+    within('#course_full_section') do
```

Comment:
We could think about using something like within `find('. govuk-grid-row)[0]` through to `find('. govuk-grid-row)[7]`